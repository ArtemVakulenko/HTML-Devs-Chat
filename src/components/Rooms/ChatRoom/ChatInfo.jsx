import React from 'react';
import PropTypes from 'prop-types';
import { StInfo, StMessageCount } from './styled.js';

const ChatInfo = ({ unreadCount }) => {
  return (
    <StInfo >
      {
        unreadCount
          ? <StMessageCount><span>{unreadCount}</span></StMessageCount>
          : null
      }
    </StInfo>
  );
};

ChatInfo.propTypes = {
  unreadCount: PropTypes.number,
};

export default ChatInfo;
