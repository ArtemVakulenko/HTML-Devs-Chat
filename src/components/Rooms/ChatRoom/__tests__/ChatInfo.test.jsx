import { mount, shallow } from 'enzyme';
import React from 'react';
import ChatInfo from '../ChatInfo';

describe('ChatInfo', () => {
  describe('havecount', () => {
    const props = { unreadCount: 1 };
    it('Should match snapshot', () => {
      const component = shallow(<ChatInfo {...props} />);
      expect(component.html()).toMatchSnapshot();
    });
    it('Should have StInfo', () => {
      const component = mount(<ChatInfo {...props} />);
      expect(component.find('styled__StInfo')).toHaveLength(1);
    });
    it('Should have StMessageCount', () => {
      const component = mount(<ChatInfo {...props} />);
      expect(component.find('styled__StMessageCount')).toHaveLength(1);
    });
    it('Should have span', () => {
      const component = mount(<ChatInfo {...props} />);
      expect(component.find('span')).toHaveLength(1);
    });
  });
  describe('withoutCount', () => {
    const props = { unreadCount: 0 };
    it('Should match snapshot', () => {
      const component = shallow(<ChatInfo {...props} />);
      expect(component.html()).toMatchSnapshot();
    });
    it('Should have StInfo', () => {
      const component = mount(<ChatInfo {...props} />);
      expect(component.find('styled__StInfo')).toHaveLength(1);
    });
    it('Should have StMessageCount', () => {
      const component = mount(<ChatInfo {...props} />);
      expect(component.find('styled__StMessageCount')).toHaveLength(0);
    });
    it('Should have span', () => {
      const component = mount(<ChatInfo {...props} />);
      expect(component.find('span')).toHaveLength(0);
    });
  });
});
