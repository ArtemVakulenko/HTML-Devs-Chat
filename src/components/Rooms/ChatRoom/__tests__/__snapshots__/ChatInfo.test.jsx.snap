// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInfo havecount Should match snapshot 1`] = `"<div class=\\"styled__StInfo-ckAQui cfpCEt\\"><div class=\\"styled__StMessageCount-fPOoeU cQHHxW\\"><span>1</span></div></div>"`;

exports[`ChatInfo withoutCount Should match snapshot 1`] = `"<div class=\\"styled__StInfo-ckAQui cfpCEt\\"></div>"`;
